server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -96,4 +86,21 @@ class AddressIndexRepository @Inject()(conf : AddressIndexConfigModule, elasticC
       )
     }
   }.map(response => PostcodeAddressFileAddresses(response.as[PostcodeAddressFileAddress], response.maxScore))
+
```

Comment:
This one has too many differences to make it factorable: index, query and the callback.
Factoring this code won't improve readability